src/OSPSuite.FuncParser/ParsedFunction.cs

```diff
@@ -110,35 +126,36 @@ public bool CaseSensitive
       public void SetVariableNames(IEnumerable<string> variableNames)
       {
          var (variableNamesArray, size) = convertToArray(variableNames);
+         if (variableNamesArray.Any(s => s==null))
+            throw new Exception("Variable name may not be null");
+
```

Comment:
Here and in all further cases where STRINGS are passed: checking for string <> null. (that is what caused the crash)
The philosophy of FuncParser is: any invalid inputs are accepted in the first place. All the inputs are evaluated only when `Parse()` or `CalcExpression()` or `GetDimensionInfo()` are called. 
However passing a null strings apparently crashes (e.g. adding null to an std string vector).

Another possibility would be to replace null strings with empty strings in C++ part, but I think passing nulls is absolutely nonsense and should be considered as  an error in the calling app (e.g. in this case in MoBi).